---
 Fixing passing any kind of argument to --with-smclient

Problem with configure.in of the tarball is that it does not seem to work at all with any --with-smclient variation.

 configure.in |   23 +++++++++++++++++------
 1 file changed, 17 insertions(+), 6 deletions(-)

Index: b/configure.in
===================================================================
--- a/configure.in
+++ b/configure.in
@@ -226,18 +226,29 @@ GDK_TARGET="$($PKG_CONFIG --variable tar
 SMCLIENT_PKGS=
 AC_MSG_CHECKING([which smclient backend to use])
 AC_ARG_WITH([smclient],
-  [AS_HELP_STRING([--with-smclient],[which smclient backend to use (xsmp/win32/quartz)])],
+  [AS_HELP_STRING([--with-smclient],[which smclient backend to use @<:@xsmp/win32/quartz/auto@:>@])],
   [],
-  [case "$GDK_TARGET" in
-    x11) with_smclient=xsmp SMCLIENT_PKGS="sm ice" ;;
+  [$with_smclient=auto])
+
+
+if test "$with_smclient" != "no"; then
+
+  if test "$with_smclient" != "auto" -a "$with_smclient" != "yes"; then
+    GDK_TARGET="$with_smclient"
+  fi
+
+  case "$GDK_TARGET" in
+    x11|xsmp) with_smclient=xsmp SMCLIENT_PKGS="sm ice" ;;
     win32|quartz) with_smclient=$GDK_TARGET ;;
     *) with_smclient=no ;;
-   esac])
-AC_MSG_RESULT([$with_smclient])
+  esac
+
+  AC_MSG_RESULT([$with_smclient])
 
-if test "$with_smclient" != "no"; then
   PKG_CHECK_MODULES([SMCLIENT],[$SMCLIENT_PKGS])
   AC_DEFINE([WITH_SMCLIENT],[1],[Define if smclient is enabled])
+else
+  AC_MSG_RESULT([$with_smclient])
 fi
 
 AM_CONDITIONAL([WITH_SMCLIENT],[test "$with_smclient" != "no"])
